### PR TITLE
[EXPERIMENT][WIP] Add OpenVINO support for MolmoWeb-4B (molmo2) VLM

### PR DESCRIPTION
**Pull request by mlukasze**
# What does this PR do?

Adds OpenVINO export and inference support for [`allenai/MolmoWeb-4B`](https://huggingface.co/allenai/MolmoWeb-4B) and the **Molmo2** model family (`model_type="molmo2"`, text sub-model `model_type="molmo2_text"`).

Molmo2 is a multimodal VLM (SigLIP-2-style ViT + pooling adapter + Qwen3-like LLM with bidirectional image-token attention driven by `token_type_ids`), architecturally close to Gemma3. The model is exported as three OpenVINO subgraphs:

- **vision_embeddings** — `Molmo2VisionBackbone(images, pooled_patches_idx) → (last_hidden_state, valid_token_mask)`
- **text_embeddings** — token embedding (`wte`)
- **language_model** — decoder + LM head, stateful KV-cache

## Installation instructions

```sh
pip install git+https://github.com/mlukasze/optimum-intel.git@enable/allenai-MolmoWeb-4B
pip install -U openvino openvino-tokenizers nncf
pip install "transformers>=4.55,<5.0" einops torchvision accelerate pillow
```

## Exporting cmd-line

```sh
optimum-cli export openvino -m allenai/MolmoWeb-4B ov_molmoweb4b_fp16 \
    --task image-text-to-text --trust-remote-code
```

INT8 / INT4 weight compression:

```sh
optimum-cli export openvino -m allenai/MolmoWeb-4B ov_molmoweb4b_int8 \
    --task image-text-to-text --weight-format int8 --trust-remote-code

optimum-cli export openvino -m allenai/MolmoWeb-4B ov_molmoweb4b_int4 \
    --task image-text-to-text --weight-format int4 --trust-remote-code
```

## Inference script

```python
from PIL import Image
import requests
from optimum.intel import OVModelForVisualCausalLM
from transformers import AutoProcessor

model_dir = "ov_molmoweb4b_fp16"  # or int8 / int4

processor = AutoProcessor.from_pretrained(model_dir, trust_remote_code=True)
model = OVModelForVisualCausalLM.from_pretrained(model_dir, trust_remote_code=True)

image_url = "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/tasks/car.jpg"
image = Image.open(requests.get(image_url, stream=True).raw).convert("RGB")

conversation = [
    {"role": "user", "content": [
        {"type": "image"},
        {"type": "text", "text": "Describe this image."}
    ]}
]
text_prompt = processor.apply_chat_template(
    conversation, add_generation_prompt=True, tokenize=False
)
inputs = processor(text=text_prompt, images=[image], return_tensors="pt")

outputs = model.generate(**inputs, max_new_tokens=200, do_sample=False)
response = processor.decode(
    outputs[0][inputs["input_ids"].shape[1]:], skip_special_tokens=True
)
print(response)
```

## Validation (real 4B weights, OpenVINO 2026.2.1 / transformers 4.57.3)

**Export** — all three subgraphs succeed at FP16, INT8, and INT4 (group_size=128).

**Parity — OV FP16 (CPU) vs PyTorch FP32**

| Metric | Value |
| :-- | :-- |
| Logits cosine similarity (mean) | 0.999999 |
| Argmax match | 99.10 % |
| Generated text | identical to PyTorch reference ✅ |

**Inference — TTFT / ITL (ms), 32 new tokens, single image (Intel Core Ultra 9 285K + Arc Pro B60)**

| Precision | CPU | iGPU | Arc Pro B60 (dGPU) |
| :-- | :-- | :-- | :-- |
| FP16 | 10 974 / 145.5 | 9 104 / 161.0 | **264 / 26.6** |
| INT8 | 5 327 / 76.7 | 6 480 / 98.4 | **231 / 15.7** |
| INT4 | 5 582 / 54.9 | 5 087 / 72.1 | **234 / 12.7** |

All 9 device × precision combinations produce correct image descriptions. dGPU TTFT is ~13–24× faster than CPU.

## Notes

- `transformers>=4.55,<5.0` required — Molmo2 uses remote code not yet compatible with transformers 5.x.
- `trust_remote_code=True` is required for both export and inference.
- NPU is not supported (stateful KV-cache LLM).

## Before submitting
- [N/A] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests?

### CODE REVIEW

**Comment by rkazants** [comment]
already opened and working https://github.com/huggingface/optimum-intel/pull/1812
